src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -674,6 +676,17 @@ ModifyPartialQuerySupported(Query *queryTree, bool multiShardQuery,
 			{
 				Assert(hasVarArgument || hasBadCoalesce);
 			}
+
+			if (ExprHasFieldStore(targetEntry->expr))
```

Comment:
We have a generic function for finding nodes called `FindNodeMatchingCheckFunction`, which takes as a parameter a function that returns true if the node matches (e.g. IsA FieldStore).